src/utils/Cypher.ts

```diff
@@ -26,19 +35,193 @@ export default class Cypher {
+        if (value && value.length > 0) {
+          allSensitiveData[`${property}`] = value;
+        }
+      }
+    }
+    return allSensitiveData;
+
+  }
+
+  public static prepareSaveSensitiveDataMigration(setting: SettingDB,
+    valueBeforeDecrypt: string[], valueAfterDecrypt: string[], valueAfterEncrypt: string[]): SettingSensitiveData {
+    const allSensitiveData: SensitiveData[] = [];
+    for (const property of setting.sensitiveData) {
+      const sensitiveData = {
+        identifier: setting.identifier,
+        path: property,
+        initialValue: {
+          encryptedValue: valueBeforeDecrypt[property],
+          key: this.cryptoSetting.formerKey
+        },
+        clearValue: valueAfterDecrypt[property],
+        migratedValue: {
+          encryptedValue: valueAfterEncrypt[property],
+          key: this.cryptoSetting.key
+        }
+      } as SensitiveData;
+      allSensitiveData.push(sensitiveData);
+    }
+
+    return {
+      id: setting.id,
+      identifier: setting.identifier,
+      path: setting.sensitiveData,
+      sensitiveData: allSensitiveData
+    };
+  }
+
+  public static async migrateSensitiveData(tenantID: string, setting: SettingDB): Promise<void> {
+    // Get sensitive data encrypted with former key
+    const valueBeforeDecrypt = this.getSensitiveDatainSetting(setting);
+    // Decrypt sensitive data with former key
+    this.decryptSensitiveDataInJSON(setting, this.cryptoSetting.formerKey);
+    // Get sensitive data in clear format
+    const valueAfterDecrypt = this.getSensitiveDatainSetting(setting);
+    // Encrypt sensitive data with new key
+    this.encryptSensitiveDataInJSON(setting, this.cryptoSetting.key);
+    // Get sensitive data encrypted with new key
+    const valueAfterEncrypt = this.getSensitiveDatainSetting(setting);
+
+    // Save setting with sensitive data encrypted with new key
+    await SettingStorage.saveSettings(tenantID, setting);
+
+    // Prepare sensitive data migration
+    const settingSensitiveData = this.prepareSaveSensitiveDataMigration(setting, valueBeforeDecrypt, valueAfterDecrypt, valueAfterEncrypt);
+    // Save sensitive data migration of setting
+    await SensitiveDataMigrationStorage.saveSensitiveData(tenantID, settingSensitiveData);
+  }
+
+  public static async getMigrateSettingsIdentifiers(tenantID: string): Promise<string[]> {
+    const migration = await SensitiveDataMigrationStorage.getSensitiveDataMigration(tenantID);
+    return migration?.settingSensitiveData?.map((setting) => setting.identifier);
+  }
+
+  public static async migrateAllSensitiveData(tenantID: string, settings: SettingDB[]): Promise<void> {
+    try {
+      const identifiers = await this.getMigrateSettingsIdentifiers(tenantID);
+      // For each setting that contains sensitive data, migrate that data
+      for (const setting of settings) {
+        if (!identifiers?.includes(setting.identifier)) {
+          await this.migrateSensitiveData(tenantID, setting);
```

Comment:
You can put the content of this method here.
Moreover migrateSensitiveData is static and you use this and it's not public but should be private.